### PR TITLE
JBPM-5181: unified code that finds IDiagramProfile

### PR DESCRIPTION
**Pull request by jomarko**
Not ready for merge. I have to add tests.

### CODE REVIEW

**Comment by jomarko** [comment]
I added some tests, @jlindop @tsurdilo , please, can you review this PR?

**Comment by jlindop** [comment]
Looks good.
@tsurdilo This PR makes "jbpm" the default profile when looking for a profile. I assume this is correct - can you confirm?

**Comment by tsurdilo** [comment]
@jlindop @jomarko The idea of "profile" was that we can work with different types of diagram shapes given a specific profile passed to designer. The reality was always that the only profile we have implemented was the jbpm profile which maps to the bpmn2 stencil set etc. 

I would suggest that we don't do the constant 

if(profileName == null || profileName.trim().isEmpty()) {
profileName = "jbpm";
}

or at least do not hardcode "jbpm" everywhere.
I think also that a real solution given that most likely the oryx designer will not implement other profiles we remove the need to pass the profile everywhere and have the idiagramprofile use the default jbpm when its being asked for. Thats probably more change and more testing so if you guys want this intermediate solution in the meantime then OK but at least remove the constant checks for it.

**Comment by jomarko** [comment]
@tsurdilo Thanks for your comments. I extracted the given code snippet into the static method. I like my PR now much more.

I was investigating a little bit usage of IDiagramProfile implementations. But your words hold true. It is a quite complex task to remove passing implementation of IDiagramProfile. I prefer to do it in another JIRA. What do you think @jlindop @tsurdilo ?

**Comment by jlindop** [comment]
@jomarko I agree that it makes sense to do any more work with IDiagramProfile in another JIRA.
+1

**Comment by jlindop** [comment]
@jomarko I agree that it makes sense to do any more work on IDiagramProfile in another JIRA.
+1

**Comment by jlindop** [comment]
Not sure if my +1 yesterday was recognized - late yesterday github was not behaving normally, so here it is again:
+1

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/CalledElementServletTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/CalledElementServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/FileStoreServletTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/FileStoreServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/ProcessDiffServiceServletTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/ProcessDiffServiceServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/StencilPatternsServletTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/StencilPatternsServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/SyntaxCheckerServletTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/SyntaxCheckerServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/ThemeServletTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
+package org.jbpm.designer.web.server;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/ThemeServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/MenuServletTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
+package org.jbpm.designer.web.server.menu;
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/MenuServletTest.java]: Missing license.. please add before this is merged.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/connector/AbstractConnectorServletTest.java [hunk 8]
```diff
@@ -2,17 +2,60 @@
```
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/connector/AbstractConnectorServletTest.java]: Missing license.. please add before this is merged.

**Comment by manstis** [comment]
A few missing licenses.. could you please check any other applicable files too?

**Comment by jomarko** [comment]
@manstis licenses added into new files.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/connector/AbstractConnectorServletTest.java (hunk 8 above)
Comment (manstis) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/connector/AbstractConnectorServletTest.java]: @jomarko Sorry.. still missing a license in ```AbstractConnectorServletTest```

Comment (jomarko) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/server/menu/connector/AbstractConnectorServletTest.java]: @manstis no problem. It was my inattention.

**Comment by manstis** [comment]
@jomarko Thanks

I'll merge when green ( @jlindop has already +1'ed)

**Comment by jomarko** [comment]
@manstis would you mind review #346 ?

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**